pkg/server/commands/reverseexpand/reverse_expand_test.go

```diff
@@ -443,3 +447,277 @@ func TestReverseExpandIgnoresInvalidTuples(t *testing.T) {
+				Threshold:    200,
+				MaxThreshold: 200,
+			}),
+		)
+		mockThrottler.EXPECT().Throttle(gomock.Any()).Times(0)
+		dispatchCountValue := uint32(190)
+		metadata := NewResolutionMetadata()
+		metadata.DispatchCounter.Store(dispatchCountValue)
+
+		reverseExpandQuery.throttle(ctx, metadata)
+	})
+
+	t.Run("above_threshold_should_call_throttle", func(t *testing.T) {
+		mockThrottler := mocks.NewMockThrottler(mockController)
+		reverseExpandQuery := NewReverseExpandQuery(
+			mockDatastore,
+			ts,
+			WithDispatchThrottlerConfig(threshold.Config{
+				Throttler:    mockThrottler,
+				Threshold:    200,
+				MaxThreshold: 200,
+			}),
+		)
+		mockThrottler.EXPECT().Throttle(gomock.Any()).Times(1)
+		dispatchCountValue := uint32(201)
+		metadata := NewResolutionMetadata()
+		metadata.DispatchCounter.Store(dispatchCountValue)
+
+		reverseExpandQuery.throttle(ctx, metadata)
+	})
+
+	t.Run("zero_max_should_interpret_as_default", func(t *testing.T) {
+		mockThrottler := mocks.NewMockThrottler(mockController)
+		reverseExpandQuery := NewReverseExpandQuery(
+			mockDatastore,
+			ts,
+			WithDispatchThrottlerConfig(threshold.Config{
+				Throttler:    mockThrottler,
+				Threshold:    200,
+				MaxThreshold: 0,
+			}),
+		)
+		mockThrottler.EXPECT().Throttle(gomock.Any()).Times(0)
+		dispatchCountValue := uint32(190)
+		metadata := NewResolutionMetadata()
+		metadata.DispatchCounter.Store(dispatchCountValue)
+
+		reverseExpandQuery.throttle(ctx, metadata)
+	})
+
+	t.Run("dispatch_should_use_request_threshold_if_available", func(t *testing.T) {
+		mockThrottler := mocks.NewMockThrottler(mockController)
+		reverseExpandQuery := NewReverseExpandQuery(
+			mockDatastore,
+			ts,
+			WithDispatchThrottlerConfig(threshold.Config{
+				Throttler:    mockThrottler,
+				Threshold:    0,
+				MaxThreshold: 210,
+			}),
+		)
+		mockThrottler.EXPECT().Throttle(gomock.Any()).Times(1)
+		dispatchCountValue := uint32(201)
+		ctx := context.Background()
+		ctx = threshold.ContextWithDispatchThrottlingThreshold(ctx, 200)
+		metadata := NewResolutionMetadata()
+		metadata.DispatchCounter.Store(dispatchCountValue)
+
+		reverseExpandQuery.throttle(ctx, metadata)
+	})
+
+	t.Run("should_respect_max_threshold", func(t *testing.T) {
+		mockThrottler := mocks.NewMockThrottler(mockController)
+		reverseExpandQuery := NewReverseExpandQuery(
+			mockDatastore,
+			ts,
+			WithDispatchThrottlerConfig(threshold.Config{
+				Throttler:    mockThrottler,
+				Threshold:    200,
+				MaxThreshold: 300,
+			}),
+		)
+		mockThrottler.EXPECT().Throttle(gomock.Any()).Times(1)
+		dispatchCountValue := uint32(301)
+		ctx := context.Background()
+		ctx = threshold.ContextWithDispatchThrottlingThreshold(ctx, 1000)
+		metadata := NewResolutionMetadata()
+		metadata.DispatchCounter.Store(dispatchCountValue)
+
+		reverseExpandQuery.throttle(ctx, metadata)
+	})
+}
+
+func TestReverseExpandDispatchCount(t *testing.T) {
+	ds := memory.New()
+	t.Cleanup(ds.Close)
```

Comment:
Whitespace in this top-level please. It's pretty crammed. Every line of code is collapsed ontop of the previous most line. It's hard to decypher. 

```
ds := memory.New()
t.Cleanup(ds.Close)

ctx := ...

ctrl := gomock.NewController(t)
...
```